src/components/Filter.js

```diff
@@ -0,0 +1,50 @@
+import "./Filter.css";
+import { FaSearch } from "react-icons/fa";
+
+export default function Filter({ onStatusFilterChange, onNameFilterChange }) {
+  function handleFilterSelection(event) {
+    event.preventDefault();
+    const filterValueStatus = event.target.value;
+    onStatusFilterChange(filterValueStatus);
+  }
+  function handleNameFilter(event) {
+    event.preventDefault();
+    const filterValueName = event.target.charactername.value;
+    onNameFilterChange(filterValueName);
+  }
+
+  return (
+    <div className="filter-form-container">
+      <form onSubmit={handleNameFilter} className="filter-form">
+        <div className="name-search-container">
+          <input
+            type="text"
+            name="charactername"
+            id="charactername"
+            placeholder="Search for your favourite character"
+            className="search-bar-input"
+            aria-label="name search input"
+          />
+          <button
+            type="submit"
+            className="filter-form-button"
+            aria-label="name search submit "
+          >
+            <FaSearch className="mag-glass" />
+          </button>
+        </div>
+      </form>
+      <select
+        onClick={handleFilterSelection}
```

Comment:
`onChange` might be a better choice here, since it only fires after the user has made a selection